### PR TITLE
[PROF-9825] Add source code integration step to profiling setup docs

### PR DESCRIPTION
**Pull request by PyvesB**
### What does this PR do? What is the motivation?

Source code integration is an important feature that makes profiling flame graphs more intuitive and easier to explore. Encouraging users to set it up at the same time as they're configuring the profiler is desirable. Profiling currently supports the integration for Java, Python, Node.js, Go and Ruby.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Review by mbeaudru — APPROVED** [review]

**Comment by aliciascott** [comment]
Thanks! I have created an [editorial card ](https://datadoghq.atlassian.net/browse/DOCS-8078)for our team to review further

File: content/en/profiler/enabling/go.md [hunk 1]
```diff
@@ -77,35 +77,37 @@ To begin profiling applications:
 
 4. Optional: Enable the [timeline feature][7] (beta), see [prerequisites][8].
 
-5. After a minute or two, visualize your profiles in the [Datadog APM > Profiler page][9].
+5. Optional: Set up [Source Code Integration][9].
```
Comment (brett0000FF) [inline content/en/profiler/enabling/go.md:80]:
```suggestion
5. Optional: Set up [Source Code Integration][9] to connect your telemetry with your Git repositories.
```
I think it's worth adding this detail so users have a reason to check it out!

File: content/en/profiler/enabling/java.md [hunk 2]
```diff
@@ -147,7 +147,9 @@ When the service binary is built, you can use environment variables to enable an
 
    **Note**: The `-javaagent` argument needs to be before `-jar`. This adds it as a JVM option rather than an application argument. For example, `java -javaagent:dd-java-agent.jar ... -jar my-service.jar -more-flags`. For more information, see the [Oracle documentation][6].
 
-4. After a minute or two, you can visualize your profiles on the [Datadog APM > Profiling page][7].
+4. Optional: Set up [Source Code Integration][7].
```
Comment (brett0000FF) [inline content/en/profiler/enabling/java.md:150]:
```suggestion
4. Optional: Set up [Source Code Integration][7] to connect your telemetry with your Git repositories.
```

File: content/en/profiler/enabling/nodejs.md [hunk 3]
```diff
@@ -75,11 +75,13 @@ const tracer = require('dd-trace/init')
 {{% /tab %}}
 {{< /tabs >}}
 
-4. A minute or two after starting your Node.js application, your profiles will show up on the [APM > Profiler page][4].
+4. Optional: Set up [Source Code Integration][4].
```
Comment (brett0000FF) [inline content/en/profiler/enabling/nodejs.md:78]:
```suggestion
4. Optional: Set up [Source Code Integration][4] to connect your telemetry with your Git repositories.
```

File: content/en/profiler/enabling/python.md [hunk 4]
```diff
@@ -76,7 +76,9 @@ To automatically profile your code, set the `DD_PROFILING_ENABLED` environment v
 
 See [Configuration](#configuration) for more advanced usage.
 
-After a couple of minutes, visualize your profiles on the [Datadog APM > Profiler page][4].
+Optionally set up [Source Code Integration][4].
```
Comment (brett0000FF) [inline content/en/profiler/enabling/python.md:79]:
```suggestion
Optionally, set up [Source Code Integration][4] to connect your telemetry with your Git repositories.
```

**Review by brett0000FF — APPROVED** [review]
Thanks! Looks good. Before I merge, could you take a look at my suggestions? I think it's worth a quick note about what Source Code Integration is for users.

**Comment by PyvesB** [comment]
@brett0000FF thanks for the review! I find the word "telemetry" a little vague in the context of Continuous Profiler, we don't use it in the product's docs or in our UI. I've opted for "profiling data". What do you think?

**Comment by brett0000FF** [comment]
> I find the word "telemetry" a little vague in the context of Continuous Profiler, we don't use it in the product's docs or in our UI. I've opted for "profiling data". What do you think?

@PyvesB - Good point! Profiling data sounds good to me. For some APM docs, we use "observability data" to reference log/metric/trace data, so your suggestion is in line with that too.

**Comment by PyvesB** [comment]
@brett0000FF great! In that case, this is ready from my perspective, feel free to merge it (I don't have the rights) if it's good from your side as well. :)

**Pull request merged**